src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -176,6 +177,31 @@ public ResponseEntity<Map<String, FileType>> getFiles(@PathVariable Long partici
     @GetMapping(value = "/repository/{participationId}/files-content/{commitId}", produces = MediaType.APPLICATION_JSON_VALUE)
     @EnforceAtLeastInstructor
     public ResponseEntity<Map<String, String>> getFilesAtCommit(@PathVariable long participationId, @PathVariable String commitId) {
+        return getMapResponseEntityForFilesAtCommit(participationId, commitId);
+    }
+
+    /**
+     * GET /repository/{participationId}/files/{commitId} : Gets the files of the repository with the given participationId at the given commitId.
+     * The commit details view is also for students, so we enforce at least student access rights.
+     *
+     * @param participationId the participationId of the repository we want to get the files from
+     * @param commitId        the commitId of the repository we want to get the files from
+     * @return a map with the file path as key and the file content as value
+     */
+    @GetMapping(value = "/repository/{participationId}/files-content-commit-details/{commitId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @EnforceAtLeastStudent
+    public ResponseEntity<Map<String, String>> getFilesAtCommitForCommitDetails(@PathVariable long participationId, @PathVariable String commitId) {
+        return getMapResponseEntityForFilesAtCommit(participationId, commitId);
+    }
```

Comment:
Why do we need a second endpoint here that does the same as the endpoint above? The only difference is that this one is also accessible for students. Either add some specific behavior to this endpoint (e.g. more access right checks) or change the acces level for the endpoint above.